ete3/tools/ete_build_lib/task/iqtree.py

```diff
@@ -29,8 +29,8 @@ def __init__(self, nodeid, alg_phylip_file, constrain_id, model,
             base_args["-st"] = "AA" if seqtype == "aa" else "DNA"            
         else:
             if conf[confname]["-st"].startswith("CODON"):
-                if seqtype == "aa" or "aa" not in GLOBALS["seqtypes"]:
-                    raise ConfigError("IQTREE CODON models require a codon alignmen.\nProvide nucleotide sequences with '-n' and set '--nt-switch-thr 0.0' to ensure codon alignments.")
+                if seqtype == "aa" or "aa" in GLOBALS["seqtypes"]: 
```

Comment:
I think it was right in the original version. If codon models are selected, we need to check that alg was done with aa and then translated back into nts. 

So, here we expect seqtype = 'nt' and 'aa' in GLOBLALS. 

if seqtype is aa or is nt but aa not in globals, raise the Error (which I think is what the IF is covering)